### PR TITLE
added user information to change chat model from continue

### PR DESCRIPTION
**Pull request by SachinS10-lab**
- Added one info card on top to close continue's welcome page.
- Added another info card on bottom to change Continue's default chat model to our's.
- If keep existing configuration selected it won't show.

### CODE REVIEW

**Comment by SachinS10-lab** [comment]
As per the James suggestion, I implemented on the info card.

File: webviews/src/App.tsx [hunk 1]
```diff
@@ -264,6 +272,18 @@ function App() {
         <div className="final-setup-group">
           <button className="install-button" onClick={handleSetupGraniteClick} disabled={serverStatus !== ServerStatus.started || !enabled || isKeepExistingConfigSelected}>Setup Granite Code</button>
         </div >
+        {chatModel ? (
```
Comment (fbricon) [inline webviews/src/App.tsx:275]: we can detect if there are existing models in ~/.continue/config.json (see https://github.com/redhat-developer/vscode-granite/blob/main/src/configureAssistant.ts#L121). If there are none or there's only 1 model and it's the same as the selected model, then no need to display this warning.

Comment (SachinS10-lab) [reply webviews/src/App.tsx:275]: Sure. I will change that.

**Comment by fbricon** [comment]
I think I would prefer we showed a single card with those 2 blocks, at the bottom, but only after the setup has completed.

**Comment by SachinS10-lab** [comment]
Hi @fbricon,
I'm sharing a screenshot of the User Information Popup. Currently, I have implemented a closeable popup whenever our chat model is installed, and it will also display after installing our chat model. 

image

---------

image

is this okay?

Also I'm unable to fetch details from ~/.continue/config.json due to some permission issue. But now showing the user information popup is working as expected.

**Comment by SachinS10-lab** [comment]
> define permission issue please

May be I'm doing wrong, when I'm trying to fetch config file using this command ( const configFile = path.join(os.homedir(), ".continue/config.json"); ) it shows like this " Error checking chat model: TypeError: os.homedir is not a function. at checkChatModel (App.tsx:217:39) ".


I'll push my current change. In current code also checking the installed models and shows the popup.

**Comment by fbricon** [comment]
I assume you tried accessing the continue config file from the react webview directly, whcih is not possible. Webviews are sandboxed. you need to send information from the vscode backend to the webview frontend via messaging. See for example https://github.com/redhat-developer/vscode-granite/blob/main/src/panels/setupGranitePage.ts#L233-L238

**Comment by SachinS10-lab** [comment]
Hi @fbricon ,
I have changed custom popup to vscode warning message. Now whenever  a selected chat model is configured, it shows the message like in the below screenshot.  is this okay?


image

**Comment by fbricon** [comment]
I'd merge the existing information popup with the new one (information, not warning!): "Continue configuration complete. Select the [granite model] from Continue's chat model dropdown."

**Comment by fbricon** [comment]
also please resolve the conflict

**Comment by SachinS10-lab** [comment]
> I'd merge the existing information popup with the new one (information, not warning!): "Continue configuration complete. Select the [granite model] from Continue's chat model dropdown."

sounds good. then I need to add only two line of code in the configureAssistant.ts file. Should I show the other information message (" Upon launching Continue.dev, a welcome screen will be displayed.Make sure to close it before setting up IBM Granite Code Models. ")?

**Comment by fbricon** [comment]
yes the other message should be shown if we can infer Continue's onboarding page will be shown. These days you can check if there's a single model in the config file, where "provider"=== "anthropic" && "apiKey" === ""

**Comment by SachinS10-lab** [comment]
is this okay?

image

**Comment by SachinS10-lab** [comment]
Hi @fbricon, 
I created a separate PR for the code change. 
PR Number: #121

**Comment by fbricon** [comment]
Closed in favour of #121

**Pull request closed without merging**